go/pkg/sysdb/metastore/db/dbcore/core.go

```diff
@@ -231,11 +266,17 @@ func GetDBConfigForTesting() DBConfig {
 }
 
 func ConfigDatabaseForTesting() *gorm.DB {
-	db, err := ConnectPostgres(GetDBConfigForTesting())
+	cfg := GetDBConfigForTesting()
+	db, err := ConnectPostgres(cfg.Address, cfg.Username, cfg.Password, cfg.Port, cfg.DBName, cfg.SslMode, cfg.MaxIdleConns, cfg.MaxOpenConns)
 	if err != nil {
 		panic("failed to connect database")
 	}
 	SetGlobalDB(db)
 	CreateTestTables(db)
 	return db
 }
+
+func ConfigReadDatabaseForTesting() *gorm.DB {
+	globalReadDB = globalDB
+	return globalReadDB
+}
```

Comment:
For testing, we set the `read_db` as the normal `db`. Notably, we need to call this method after `ConfigDatabaseForTesting` in tests.